logger/common.go

```diff
@@ -116,55 +117,98 @@ func NewInfo(containerID string, containerName string, options ...InfoOpt) *dock
 }
 
 // Start starts the actual logger.
-func (l *Logger) Start(uid int, gid int, cleanupTime *time.Duration, ready func() error) error {
-	var wg sync.WaitGroup
-	if l.Stdout != nil {
-		wg.Add(1)
-		go l.sendLogs(l.Stdout, &wg, sourceSTDOUT, uid, gid, cleanupTime)
-	}
-	if l.Stderr != nil {
-		wg.Add(1)
-		go l.sendLogs(l.Stderr, &wg, sourceSTDERR, uid, gid, cleanupTime)
+func (l *Logger) Start(
+	ctx context.Context,
+	uid int,
+	gid int,
+	cleanupTime *time.Duration,
+	ready func() error,
+) error {
+	if l.Stdout == nil || l.Stderr == nil {
+		return errors.New("no stdout/stderr pipe opened")
 	}
 
+	errGroup, ctx := errgroup.WithContext(ctx)
+	errGroup.Go(func() error {
+		err := l.sendLogs(ctx, l.Stdout, sourceSTDOUT, uid, gid, cleanupTime)
+		if err != nil {
+			stdoutErr := errors.Wrapf(err, "failed to send logs from pipe %s", sourceSTDOUT)
+			debug.SendEventsToJournal(DaemonName, stdoutErr.Error(), journal.PriErr, 1)
+			return stdoutErr
+		}
+		return nil
+	})
+	errGroup.Go(func() error {
+		err := l.sendLogs(ctx, l.Stderr, sourceSTDERR, uid, gid, cleanupTime)
+		if err != nil {
+			stderrErr := errors.Wrapf(err, "failed to send logs from pipe %s", sourceSTDERR)
+			debug.SendEventsToJournal(DaemonName, stderrErr.Error(), journal.PriErr, 1)
+			return stderrErr
+		}
+		return nil
+	})
```

Comment:
Non-blocking: since these goroutine functions are so similar, would be good to pull this out into its own function.

Reply:
Yeah that makes sense! I will combine them together in the next pr.